.github/workflows/build_and_test.yml

```diff
@@ -147,8 +147,9 @@ jobs:
             mllib-local, mllib, graphx
           - >-
             streaming, sql-kafka-0-10, streaming-kafka-0-10, streaming-kinesis-asl,
-            yarn, kubernetes, hadoop-cloud, spark-ganglia-lgpl,
-            connect, protobuf
+            kubernetes, hadoop-cloud, spark-ganglia-lgpl, protobuf
+          - >-
+            yarn, connect
```

Comment:
Hmmm ... Does separating the build help the stabilization? I am not against this change but splitting this will cause 30mins time for building. Although it runs in parallel, individual fork has some resource limitation.

Reply:
Ya, I also was reluctant. It will take `34 minutes` in total run time. The second goal is to reduce the re-trigger time. Previously, YARN/Connect/Kafka modules build a bad synergy because their failure rates are multiplied. And, I needed to re-trigger and wait over 70 ~ 90minutes for this pipeline.

![Screenshot 2024-02-14 at 16 39 19](https://github.com/apache/spark/assets/9700541/e1afe7d3-8c0c-4ac0-93d0-34c810eb6366)

Reply:
Now, the `streaming, ...` pipeline run time is down to 62m. So, the total increased overhead is around 20min.

![Screenshot 2024-02-14 at 16 47 28](https://github.com/apache/spark/assets/9700541/18ba8ad1-3295-4aff-b853-bbc30f3c830a)

Reply:
hmmm.. I think we should actually better fix the flakiness .. I am fine with this as a temporary change.

Reply:
Thanks! I hoped so too. For now, `SparkSessionE2ESuite` flakiness and `YarnClusterSuite` seems to be abandoned.

Reply:
From my side, we don't use YARN and Spark Connect.